packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -35,134 +39,143 @@ interface InMemoryAnchorConfig {
   verifySignatures?: boolean;
 }
 
+const SAMPLE_ETH_TX_HASH =
+  "bagjqcgzaday6dzalvmy5ady2m5a5legq5zrbsnlxfc2bfxej532ds7htpova";
+
+// Sequential execution of requestAnchor commands
+const requestAnchorQueue = new PQueue({ concurrency: 1 });
+
 /**
  * In-memory anchor service - used locally, not meant to be used in production code
  */
-class InMemoryAnchorService extends AnchorService {
-  private _ceramic: Ceramic
-  private _dispatcher: Dispatcher
-
-  private readonly _anchorDelay: number
-  private readonly _anchorOnRequest: boolean
-  private readonly _verifySignatures: boolean
+class InMemoryAnchorService implements AnchorService {
+  #ceramic: Ceramic;
+  #dispatcher: Dispatcher;
 
-  private _queue: Candidate[] = []
+  readonly #anchorDelay: number;
+  readonly #anchorOnRequest: boolean;
+  readonly #verifySignatures: boolean;
+  readonly #feed: Subject<AnchorServiceResponse> = new Subject();
```

Comment:
So as best I can tell `Subject` is the same as `Observable` but allows multiple subscribers instead of just one.  Why do we need to support multiple subscribers here?

Reply:
Okay, I see now, it's because we send all the anchor requests through this single feed, then filter the global feed to individual per-request observables that we return from `requestAnchor`.  So then why have this single unified `#feed` at all?  The ethereum-anchor-service doesn't do this - why can't the in-memory anchor service do the same and just have a bunch of totally separate per-request observables without ever serializing them into a single feed?

Reply:
The logic here is different. Unlike independent requests for ethereuem, all of these requests are triggered by  single call to `#anchor`. That's why.

Reply:
Ah right, that makes sense.  The in-memory anchor service supports `anchorOnRequest:false` which the eth anchor service does not.